nft-client/README.md

```diff
@@ -0,0 +1,102 @@
+# NFT-client
+
+This is an example of an off-chain application which integrates with a smart contract on chain. The application is a client for working with a specific implementation of a Non-fungible token(NFT) smart contract.
+The smart contract should follow the CTS1 specification, but some additional assumptions are made for how the contract serializes the state and the existence of a "mint" contract function.
+
+The smart contract is called "CTS1-NFT" and can be found [here](https://github.com/Concordium/concordium-rust-smart-contracts/tree/main/examples).
+
+Most of the functionality needs access to the GRPC API of a running concordium-node.
+Some functionality of the client additionally depends on the concordium-node to have the transaction logging enabled and access to
+the PostgreSQL database with transaction logs.
+
+Run the following for a description of the functionality:
+
+```
+nft-client --help
+```
+
+## Commands
+
+The NFT-client supports the following commands:
+
+### Minting NFTs
+
+Call the smart contract function for minting a number of NFTs with provided token IDs.
+This command will result in a transaction on the blockchain and requires account keys, see section for setting up account keys.
+
+Notice only the smart contract will only allow the contract owner to call the mint function.
+
+#### Example:
+
+To mint two NFTs with tokenID `0a` and `0abe` (hex encoded) we can run the following command
+```
+nft-client mint --contract "<54,0>" --sender key-test.json 0a 0abe
```

Comment:
To recap the discussion we had the other day about this notation:

The main issue is for the `transfer` command. 
Since the `--from` and `--to` could be contract addresses, this command could potentially need 3 contract addresses.
All 3 should have an optional subindex.

One approach is to have the additional optional arguments `--contract-subindex`, `--from-subindex` and `--to-subindex`.